### ISSUE TITLE
OpenTracingShim context propagation not working with agent

### ISSUE THREAD

**Issue by fuleow** [open-telemetry/opentelemetry-java-instrumentation#1537]
I have a custom library that uses OpenTracing and it needs to propagate context headers. I am trying to use OpenTelemetry's OpenTracing shim to interface with this library's tracing. Everything works correctly in my unit test when I set up the OpenTelemetry tracer manually

```
OpenTelemetrySdk.getTracerManagement().addSpanProcessor(SimpleSpanProcessor.newBuilder(new LoggingSpanExporter()).build());
ContextPropagators propagators = DefaultContextPropagators.builder().addTextMapPropagator(HttpTraceContext.getInstance()).build();
OpenTelemetry.setPropagators(propagators);
Tracer tracer = TraceShim.createTracerShim();
```

However, when I deploy my service running the agent the propagation does not work.

```
ContextPropagators propagators = DefaultContextPropagators.builder().addTextMapPropagator(HttpTraceContext.getInstance()).build();
OpenTelemetry.setPropagators(propagators);
Tracer tracer = OpenTelemetry.getTracer("com.example.service");

// OpenTracing shim
GlobalTracer.registerIfAbsent(TraceShim.createTracerShim());
```

I realized later that it appears setting a propagator doesn't do anything when the agent is used because it always returns an `ApplicationContextPropagator`
https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/v0.9.0/instrumentation/opentelemetry-api-beta/src/main/java/io/opentelemetry/javaagent/instrumentation/opentelemetryapi/OpenTelemetryApiInstrumentation.java#L82

Debugging further it makes it to the `ApplicationTextMapPropagator` but the agentContext is null and the injection fails:

https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/v0.9.0/instrumentation/opentelemetry-api-beta/src/main/java/io/opentelemetry/javaagent/instrumentation/opentelemetryapi/context/propagation/ApplicationTextMapPropagator.java#L58
 
I am wondering if there is any documentation on how to properly set up a tracer in the application to do manual tracing in conjunction with the agent.

Also does the agent support the OpenTracing shim?

**Comment by fuleow** [comment]
This looks like related to the context issues that affected grpc server tracing. Downgrading to the 0.8.0 agent fixes both of these problems.

**Comment by trask** [comment]
hi @fuleow! there have been significant improvements/fixes in context propagation interop in 0.10.0, which we are hoping to release end of this week, would be great if you could re-test after 0.10.0 is released and let us know how that goes

**Comment by fuleow** [comment]
Tested in 0.11 and this is fixed, thank you
